cdap-docs/admin-manual/source/security/secure-storage.rst

```diff
@@ -4,12 +4,49 @@
 
 .. _admin-secure-storage:
 
-==========================
-Configuring Secure Storage
-==========================
+==============
+Secure Storage
+==============
 
-**Secure Storage** allows users to store sensitive information such as passwords in an encrypted manner.
+Applications can need controlled access to sensitive data such as passphrases, cryptographic keys, access tokens, and
+passwords. This data is usually small in size, but needs to be stored and managed in a secure manner.
+*Secure Storage* allows users to store such sensitive information in an secure and encrypted manner. Data is encrypted
+upon submission to CDAP (via RESTful or programmatic APIs) and is decrypted upon retrieval.
 
 **Note:** In CDAP 3.5.0, encryption and decryption of the contents only happens at the
 secure store, not while the data is transitting to the secure store. In a later version of
 CDAP, all transport involving secure keys will be secured using SSL.
+
+Secure Storage Format
+---------------------
+An entry in secure storage consists of:
+
+- **Key**: An alias for the entry, also referred to as a :term:`Secure Key <secure key>`.
+  Data is stored against the provided key and can be retrieved using the same key. 
+  Key must be of the :ref:`Alphanumeric Character Set <supported-characters>`, contain *only
+  lowercase* characters, and should start with a letter.
+- **Data**: The data which is to be stored in a secure an encrypted manner. This could be a passphrase, 
+  cryptographic key, access token, or any other data that needs to be stored securely.
```

Comment:
I could not find a size limitation mentioned any where for either KMS or the java keystore.